pkg/fluentd/fluentd_test.go

```diff
@@ -90,6 +91,45 @@ func TestCreateFluentdContainer(t *testing.T) {
+	os.Unsetenv("SINGLE_SYSTEM_VMI")
+
+	assertion.Equal("fluentd", fluentd.Name, "Fluentd container name not equal to expected value")
+	assertion.Equal(2, len(fluentd.Args), "Fluentd container args count not equal to expected value")
+	assertion.Equal("-c", fluentd.Args[0], "Fluentd container arg not equal to expected value")
+	assertion.Equal("/etc/fluent.conf", fluentd.Args[1], "Fluentd container arg not equal to expected value")
+	assertion.Equal(7, len(fluentd.Env), "Fluentd container envs count not equal to expected value")
+	assertion.Equal("APPLICATION_NAME", fluentd.Env[0].Name, "Fluentd container envs name not equal to expected value")
+	assertion.Equal("test-component", fluentd.Env[0].Value, "Fluentd container envs value not equal to expected value")
+	assertion.Equal("FLUENTD_CONF", fluentd.Env[1].Name, "Fluentd container envs name not equal to expected value")
+	assertion.Equal("fluentd.conf", fluentd.Env[1].Value, "Fluentd container envs value not equal to expected value")
+	assertion.Equal("FLUENT_ELASTICSEARCH_SED_DISABLE", fluentd.Env[2].Name, "Fluentd container envs name not equal to expected value")
+	assertion.Equal("true", fluentd.Env[2].Value, "Fluentd container envs value not equal to expected value")
+	assertion.Equal("ELASTICSEARCH_HOST", fluentd.Env[3].Name, "Fluentd container envs name not equal to expected value")
+	assertion.Equal("vmi-"+constants.VmiSystemBindingName+"-es-ingest.verrazzano-system.svc.cluster.local", fluentd.Env[3].Value, "Fluentd container envs value not equal to expected value")
+	assertion.Equal("ELASTICSEARCH_PORT", fluentd.Env[4].Name, "Fluentd container envs name not equal to expected value")
+	assertion.Equal("9200", fluentd.Env[4].Value, "Fluentd container envs value not equal to expected value")
+	assertion.Equal("ELASTICSEARCH_USER", fluentd.Env[5].Name, "Fluentd container envs name not equal to expected value")
+	assertion.Equal(constants.VmiSecretName, fluentd.Env[5].ValueFrom.SecretKeyRef.Name, "Fluentd container envs value from secret name not equal to expected value")
+	assertion.Equal("username", fluentd.Env[5].ValueFrom.SecretKeyRef.Key, "Fluentd container envs value from secret key not equal to expected value")
+	assertion.Equal(true, *fluentd.Env[5].ValueFrom.SecretKeyRef.Optional, "Fluentd container envs value from secret optional not equal to expected value")
+	assertion.Equal(constants.VmiSecretName, fluentd.Env[5].ValueFrom.SecretKeyRef.Name, "Fluentd container envs value from secret name not equal to expected value")
+	assertion.Equal("password", fluentd.Env[6].ValueFrom.SecretKeyRef.Key, "Fluentd container envs value from secret key not equal to expected value")
+	assertion.Equal(true, *fluentd.Env[6].ValueFrom.SecretKeyRef.Optional, "Fluentd container envs value from secret optional not equal to expected value")
+	assertion.Equal(3, len(fluentd.VolumeMounts), "Fluentd container volume mounts count not equal to expected value")
```

Comment:
Maybe should fail if len is wrong to avoid potential out of bounds condition in subsequent checks.

Reply:
same as above